src/preferenceCenter/preferenceCenter.domcomponents.js

```diff
@@ -0,0 +1,228 @@
+            this.listenTo(this.model, 'change:attributes:checked', this.handleChecked);
+          },
+          handleChecked: function () {
+            this.el.checked = !!this.model.get('attributes').checked;
+          },
+        },
+      }),
+      dc.addType('label', {
+        extend: 'text',
+        isComponent: function (e) {
+          return 'LABEL' == e.tagName;
+        },
+        model: { defaults: { tagName: 'label', components: 'Label' } },
+      });
+    dc.addType('option', {
+      isComponent: (el) => el.tagName == 'OPTION',
+
+      model: {
+        defaults: {
+          tagName: 'option',
+          layerable: false,
+          droppable: false,
+          draggable: false,
+          highlightable: false,
+        },
+      },
+    });
+
+    const createOption = (value, name) => ({
+      type: 'option',
+      components: name,
+      attributes: { value },
+    });
+
+    dc.addType('select', {
+      extend: 'input',
+      isComponent: (el) => el.tagName == 'SELECT',
+
+      model: {
+        defaults: {
+          tagName: 'select',
+          components: [
+            createOption('email', 'Email'),
+            //createOption('opt2', 'Option 2'),
+          ],
+          traits: [
+            nameTrait,
+            {
+              name: 'options',
+              type: 'select-options',
+            },
+            requiredTrait,
+          ],
+        },
+      },
+
+      view: {
+        events: {
+          mousedown: (e) => e.preventDefault(),
+        },
+      },
+    });
+
+    const domc = this.editor.DomComponents;
+    const defaultType = domc.getType('default');
+    const defaultModel = defaultType.model;
+    const defaultView = defaultType.view;
+
+    dc.addType('date', {
+      model: defaultModel.extend(
+        {
+          defaults: {
+            ...defaultModel.prototype.defaults,
+            'custom-name': 'date',
+            tagName: 'input',
+            type: 'date',
+            attributes: {
+              'data-gjs-type': 'date',
+              'data-slot': 'date',
+            },
+            droppable: false, // Can't drop other elements inside
+            copyable: false, // Do not allow to duplicate the component
+            script: function () {
+              var input = this;
+              var initDateRange = function () {
+                var input = this;
+                const options = {
+                  singleDatePicker: true,
+                  showDropdowns: true,
+                };
+                $(input).daterangepicker(options);
+              };
+
+              if (
+                typeof daterangepicker == 'undefined' ||
+                typeof jQuery == 'undefined' ||
+                typeof moment == 'undefined'
+              ) {
+                if (typeof jQuery == 'undefined') {
+                  var jquery = document.createElement('script');
+                  jquery.src = '//cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js';
```

Comment:
I don't think it is a good idea to load the JavaScript dependencies inside the code. We should start using npm to manage our dependencies. I guess the idea of this is to only load it when needed? 

Possible option: 
Require it using npm and then load it from our codebase.
I guess things like jquery and bootstrap should be loaded anyway already in our codebase.